### PR TITLE
Fix major typo: It is "UpdateServices", not "WSUS"

### PR DESCRIPTION
**Pull request by skycommand**
The documentation has a major typo repeated all over: The name of the module in Windows 10 and Windows Server 2016 containing the WSUS cmdlets is mistakenly written as "WSUS". The correct module name is "UpdateServices".

I am attaching screenshots that prove my point. These screenshots are taken from Windows 10, Windows Server 2016 and Windows Server 2019. As you can see, the Get-Module cmdlet is invoked on them to look for installed modules called "WSUS" and "UpdateServices". Get-Module does not find any modules named "WSUS", but the UpdateServices is always available. The last screenshot also shows all exported cmdlets of the UpdateServices module, showing that indeed all sixteen involved cmdlets are there.

Please feel free to test this yourself.

Windows 10 v1909
Windows Server 2016
Windows Server 2019

In this pull request, I have tried to fix the typo by:

1. Renaming the folder containing the pertinent 17 files.
2. Renaming WSUS.md to UpdateServices.md
3. Editing the content and front matter of UpdateServices.md.

I hope I haven't missed anything.

### CODE REVIEW

**Comment by e0i** [comment]
@nenonix @JohanFreelancer9 Copy editing is needed for this PR. Thanks.

**Comment by e0i** [comment]
@dariomws306 @get-itips @tseward

Peer review needed for this PR. Thanks.

**Comment by skycommand** [comment]
> @dariomws306 @get-itips @tseward
> 
> Peer review needed for this PR. Thanks.

Huh? So, I propose renaming a group of files, and you propose proofreading their contents? Not that proofreading is anything bad, but I think these two could be performed independent of each other.

**Comment by e0i** [comment]
@traya1 (PubOps) Would you be able to advise concerning changing folder vs module names? Is it favorable in the face of possible side effects? Thanks.

**Comment by e0i** [comment]
@skycommand The PR is under investigation. Would you re-target the PR branch to `master` in the meantime? Thank you.

**Comment by skycommand** [comment]
@e0i Done.

**Comment by kenwith** [comment]
@skycommand - thank you for the detailed PR and screenshots and the work you have done on this. Your contribution and passion is very much appreciated. @get-itips shared a related issue here: https://github.com/MicrosoftDocs/windows-powershell-docs/issues/133

I have no doubt that you and everyone in this thread knows way more than me so bear with me as we get this figured out. 

We will also need to add redirects so that the old paths continue to work for blog articles and other references to the cmdlets. I believe we tried this before and ran into problems with getting it to work but let's try again. Below is guidance I found on the redirects and we will try it.


# How to redirect obsolete articles

To avoid broken links, you must redirect files that are removed, renamed, or moved to a new location. You can redirect by using a master redirect file in the repo, or by requesting site-level redirection.

## Master redirect file

You should manage redirects in a single file at the root of your repo, .openpublishing.redirection.json. Using a single file makes it easier to track which files have been redirected, and lets you remove the redirected files from the repo.

To add a redirect to .openpublishing.redirection.json, add an entry to the `redirections` array:

```json
{
    "redirections": [
        {
            "source_path": "openpublishing/docs/partnerdocs/index.md",
            "redirect_url": "/opsdocs/partnerdocs/VSTSgit-github",
            "redirect_document_id": true
        },
```

- The `source_path` is the relative link to the file you want to redirect.
- The `redirect_url` is the link to the target page. `redirect_url` can be a relative link to another file in the same docset, an absolute link to another article on docs.microsoft.com, or a URL to any web page.
- `redirect_document_id` indicates whether you would like to keep the document ID from the previous file. The default is `false`. Use `true` if you want to preserve the `ms.documentid` attribute value from the redirected article. If you preserve the document ID, data, such as page views and rankings, will be transferred to the target article.

> [!IMPORTANT]
> Using this method, you can only preserve the document ID if the target file is in the same docset as the redirected file. If you redirect multiple files to the same target file, you can only set `redirect_document_id` to `true` for one of them. If you are moving existing articles from one repository to another repository, you need to file a ticket with the OPS development team - they can assist with a programmatic solution that allows you to preserve the document ID.

After you've redirected a file via the master redirect file, you must remove the original file from the repo. Otherwise, you'll get a build warning:

`[Warning] 'path-to-file.md' would be overwritten by redirection rule configured in master redirection file .openpublishing.redirection.json. Please remove the original file to resolve this warning.`

For interactive tutorials and Microsoft Learn content, you can redirect from the `index.yml` file by substituting `index.md` in the master redirection file. 

## How to generate a master redirect file based on file-level redirects

Before master redirect files were supported, we had to redirect files individually by adding the `redirect_url` and `redirect_document_id` (optional) metadata to the YML header of the Markdown file:

```markdown
---
redirect_url: https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-authoring-pack
redirect_document_id: false
title: Gauntlet VS Code extension
author: meganbradley
ms.author: mbradley
ms.date: 05/18/2018
ms.topic: contributor-guide
---
```

This approach is no longer recommended. If you have a repo that still uses file-level redirection, you can clean it up by using the Docs Markdown VS Code extension functionality to generate a master redirect file and remove the obsolete files from the repo.

1. Install the Docs Markdown VS Code extension, either [individually](https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-markdown) or as part of the [Docs Authoring Pack](https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-authoring-pack).
1. Open your cloned repo and make sure it's up-to-date.
1. Hit F1 to open the VS Code command palette.
1. Start typing "Docs: Generate master redirect file" until you see the command, then select it.
1. The master redirection script will run on the repo. For each Markdown file it finds with the `redirect_url` attribute, it will:
   1. Add an entry for the file to .openpublishing.redirection.json. The path of the file becomes the value of `source_path`. The value of `redirect_url` in the master redirect file is the same as in the source file.
   1. If you included the `redirect_document_id` attribute in your Markdown file with the value `true`, it will add this to the master redirect file. Otherwise, it will default to `redirect_document_id` equals `false`.
   1. Move the redirected file to a folder under your default path, such as `C:\Users\your-alias\Docs Authoring\Redirects\repo-name_deleted_redirects_date-and-time`.
1. When the script completes, check the deleted redirects folder to make sure the files were removed as expected.
1. If so, submit a PR to update the repo.

You can run the script multiple times if new file-level redirects creep into the repo.

After a few runs of the script, your Redirects folder might look something like this:

redirected files

## Site-level redirection

Site-Level redirection works for large content sets to move in bulk, but is **only** available for certain scenarios.  If your redirection meets these requirements, you can request bulk redirection:

### Site-level redirection requirements

 - Must be an MSDN and MTPS to Docs redirection scenario.
 - The docsets must have different `Base Path` values displayed in the Open Publishing Portal.
 - Must be a custom redirection scenario that isn't achievable in a master redirection file.
 
If your request meets these requirements, please file a site-Level redirection request by following these steps:
 
1. Go to [https://sitehelp.microsoft.com](https://sitehelp.microsoft.com).
1. Click `Submit request or issue`.
1. Select `Redirections` for the service, then the appropriate service category.
1. Follow the instructions provided.

**Comment by skycommand** [comment]
Hello, @kenwith. 😊 That was a lot to digest but I think I've got a hang of it. Thanks for spending the time writing it.

Now, if you have a minute to spare, let's start from the top and take a look at what needs to be done together.

First, to create a redirection, I need to source URL, right? Well, here is our big problem: Of all the 17 files in my PR, none of them have their "online version" field of their front matters. `UpdateServices.md`, formerly `WSUS.md`, has a URL in somewhere else in its front matter. The URL is <http://go.microsoft.com/fwlink/?linkid=390839> If you click on it, you get a 404 error.

To further test this, I did the following:

1. Installed "RSAT: Windows Server Update Services Tools" on my home computer
2. Restarted the computer
3. Verified that the `UpdateServices` module is truly installed
4. Updated PowerShell's help via `Update-Help`
5. Run the following script: `Get-Command -Module UpdateServices | ForEach-Object {Get-Help $_.Name -Online -Verbose}`

Here is the result:

```` text
VERBOSE: Help URI: http://go.microsoft.com/fwlink/?LinkId=786992
VERBOSE: Help URI: http://go.microsoft.com/fwlink/?LinkId=786996
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287825
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287826
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287827
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287828
VERBOSE: Help URI: http://go.microsoft.com/fwlink/?LinkId=789378
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287829
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287830
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287831
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287832
VERBOSE: Help URI: http://go.microsoft.com/fwlink/?LinkId=789379
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287833
VERBOSE: Help URI: http://go.microsoft.com/fwlink/?LinkId=789380
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287834
VERBOSE: Help URI: http://go.microsoft.com/fwlink/p/?linkid=287835
````

We can probably add these links back into the .md files, but should we? First, who removed them and why? Second, as @sdwheeler mentioned in #4553, `fwlink` links can only be edited by a Microsoft internal tool.

At this point, I don't know how to proceed.

**Comment by sdwheeler** [comment]
@kenwith The redirection instructions you provided are for redirecting renamed or deleted documents in our website. That does not apply to the links that @skycommand is referring to.

@skycommand we could update the frontmatter in the MD files to include these links but those changes would not show up via `Update-Help` because there is not build pipeline to create new Help CAB files for download. This is a know issue that the docs team is working on with the PowerShell team, but we are months away from any real solution. 

The best we can do at this point is to fix the frontmatter so that it will publish correctly when we fix the build pipeline. With the fixed frontmatter, you could also do a local build as I described in my blog post: https://devblogs.microsoft.com/powershell/updating-help-for-older-versions-of-powershell/

I can possibly help with the FWLinks. Can you provide a list of which link is supposed to be for which cmdlet? I will also investigate the same on my end.

**Comment by kenwith** [comment]
@sdwheeler thanks Sean! 

@get-itips  and @dariomws306 can help too. I know they were looking into this already.

Based on my limited knowledge I think we can put the Online URL as the first "related links" entry and then Online will work. Here is some details I found: https://info.sapien.com/index.php/scripting/scripting-help/helpuri-v-helpinfouri

I know that Chris does this in Exchange, here is example: https://github.com/MicrosoftDocs/office-docs-powershell/blob/master/exchange/exchange-ps/exchange/active-directory/Add-ADPermission.md

That way we can get away from fwlink dependency.

**Comment by sdwheeler** [comment]
@kenwith and I talked. Putting the Online URL as the first "related link" is not recommended as it is not supported for the `-Online` experience and can be broken in future releases. For the MD files we need to add a link in the frontmatter. But that link should be a direct link to the article on docs.microsoft.com. The FWLinks should only be used in the cmdlet source code to bootstrap the help experience. The existing FWLinks need to be updated (internally) to fix the 404 issues.

**Comment by sdwheeler** [comment]
The pattern for the `online version:` metadata in the frontmatter is as follows:

```
online version: https://docs.microsoft.com/powershell/module/<module-name>/<cmdlet-name>?view=<version-moniker>
```

So for the `Add-WsusComputer` cmdlet the new correct link will be:

```
online version: https://docs.microsoft.com/powershell/module/UpdateServices/Add-WsusComputer?view=win10-ps
```

Note that he URL does not include a locale (en-us).

**Comment by skycommand** [comment]
Hmm... I think what we need here is to take a pragmatic approach, and I think do see a pragmatic solution here.

Currently, our bad links are as follows. (I deleted the "https://docs.microsoft.com/en-us/powershell/module" portion for brevity.)

1. /wsus/Add-WsusComputer?view=win10-ps
2. /wsus/Add-WsusDynamicCategory?view=win10-ps
3. /wsus/Approve-WsusUpdate?view=win10-ps
4. /wsus/Deny-WsusUpdate?view=win10-ps
5. /wsus/Get-WsusClassification?view=win10-ps
6. /wsus/Get-WsusComputer?view=win10-ps
7. /wsus/Get-WsusDynamicCategory?view=win10-ps
8. /wsus/Get-WsusProduct?view=win10-ps
9. /wsus/Get-WsusServer?view=win10-ps
10. /wsus/Get-WsusUpdate?view=win10-ps
11. /wsus/Invoke-WsusServerCleanup?view=win10-ps
12. /wsus/Remove-WsusDynamicCategory?view=win10-ps
13. /wsus/Set-WsusClassification?view=win10-ps
14. /wsus/Set-WsusDynamicCategory?view=win10-ps
15. /wsus/Set-WsusProduct?view=win10-ps
16. /wsus/Set-WsusServerSynchronization?view=win10-ps

We need to make them look like the following:

1. /updateservices/Add-WsusComputer?view=win10-ps
2. /updateservices/Add-WsusDynamicCategory?view=win10-ps
3. /updateservices/Approve-WsusUpdate?view=win10-ps
4. /updateservices/Deny-WsusUpdate?view=win10-ps
5. /updateservices/Get-WsusClassification?view=win10-ps
6. /updateservices/Get-WsusComputer?view=win10-ps
7. /updateservices/Get-WsusDynamicCategory?view=win10-ps
8. /updateservices/Get-WsusProduct?view=win10-ps
9. /updateservices/Get-WsusServer?view=win10-ps
10. /updateservices/Get-WsusUpdate?view=win10-ps
11. /updateservices/Invoke-WsusServerCleanup?view=win10-ps
12. /updateservices/Remove-WsusDynamicCategory?view=win10-ps
13. /updateservices/Set-WsusClassification?view=win10-ps
14. /updateservices/Set-WsusDynamicCategory?view=win10-ps
15. /updateservices/Set-WsusProduct?view=win10-ps
16. /updateservices/Set-WsusServerSynchronization?view=win10-ps

So, if we redirect all of the links in the former list to all of the links in the latter, we solve the bulk of our problem without needing to update the `fwlink` redirects or fix the build pipeline.

Now, please do tell me if I am being wrong here: The correct way is to make a `.openpublishing.redirection.json` at the root of `windows-powershell-docs`, whose content is:

```json
{
    "redirections": [
        {
            "source_path": "docset/windows/updateservices/Add-WsusComputer.md",
            "redirect_url": "/powershell/module/updateservices/Add-WsusComputer",
            "redirect_document_id": true
        },
````

File: .openpublishing.redirection.json [hunk 1]
```diff
@@ -929,6 +929,86 @@
         "source_path": "docset/virtual-directory-module/medv/new-medvworkspace.md",
         "redirect_url": "/powershell/module/Microsoft.Medv.Administration.Commands.WorkspacePackager/new-medvworkspace",
         "redirect_document_id": false
+        },
+        {
+        "source_path": "docset/windows/wsus/Add-WsusComputer.md",
+        "redirect_url": "/powershell/module/updateservices/Add-WsusComputer",
```
Comment (sdwheeler) [inline .openpublishing.redirection.json]: Need to include the version moniker at the end of the redirect_url.

"redirect_url": "/powershell/module/updateservices/Add-WsusComputer?view=win10-ps",

**Comment by sdwheeler** [comment]
#sign-off

**Comment by skycommand** [comment]
Much obliged, @sdwheeler 👍

**Pull request merged**